plugin/src/Reconciler/applyPatch.lua

```diff
@@ -88,10 +88,10 @@ local function applyPatch(instanceMap, patch)
 		}
 		local partiallyApplied = false
 
-		-- If the instance's className changed, we have a bumpy ride ahead while
-		-- we recreate this instance and move all of its children into the new
-		-- version atomically...ish.
-		if update.changedClassName ~= nil then
+		-- If the instance's className changed, or there is custom recreate logic,
+		-- we have a bumpy ride ahead while we recreate this instance and move all 
+		-- of its children into the new version atomically...ish.
+		if update.requiresRecreate then
```

Comment:
There are comments in this block of code about its caveats, notably not preserving existing properties if we end up needing to recreate an instance. It's probably not the end of the world, but some of the comments I left about this mostly occurring during Folder<->ModuleScript transitions are probably not accurate anymore.